### ISSUE TITLE
"import" keyword can not import a file

### ISSUE THREAD

**Issue by duoduo70** [Enalye/grimoire#8]
```dlang
compiler.addFile("test.gr");
```
in test.gr
```grimoire
import "a.gr"
event main() {
a();
}
```

error: unknown function

i used Grimoire wrong？ Or it is a bug?

**Comment by Enalye** [comment]
Hello,

Does the file `a.gr` exists ?
Does the function or task `a()` exists in `a.gr` and is exported ?

**Comment by duoduo70** [comment]
Yes. I created a smallest test project.

In test.d
```d
import grimoire;

void main(string[] args)
{
    GrLibrary stdlib = grLoadStdLibrary();
    GrCompiler compiler = new GrCompiler;
    compiler.addLibrary(stdlib);
    compiler.addFile("script.gr");
    GrBytecode bytecode = compiler.compile();
    if (bytecode)
    {
        GrEngine engine = new GrEngine;
        engine.addLibrary(stdlib);
        engine.load(bytecode);
        GrTask task = engine.callEvent("main");
        while (engine.hasTasks)
            engine.process();
    }
    else
    {
        import std.stdio : writeln;
        writeln(compiler.getError().prettify(GrLocale.en_US));
    }
}
```

In script.gr
```grimoire
import "a.gr"
event main() {
    print(a);
}
```

In a.gr (In the root directory with script.gr)
```grimoire
export var a = "1";
```

The shell output:
```
error: `a` is not declared
  -> /<my-dirctory>/script.gr(3,11)
  |
 3|     print(a);
  |           ^  unknown variable
```

**Comment by duoduo70** [comment]
For another example, if I create a paser bug in a.gr, the compiler will not report it to me.

**Comment by Enalye** [comment]
Ok found the issue, it was a dumb mistake on my part.

The problem was that the imported file was ignored because I simply forgot to compare the file path with already imported files, thus discarding it.

Fixed in commit 7a7752e92aae13ed70fcc902a5b0bfbfcabe8796

Please let me know if it’s correctly fixed on your end.

**Comment by duoduo70** [comment]
It work
